lib/std/elf.zig

```diff
@@ -532,17 +532,23 @@ pub const Header = struct {
         };
         const need_bswap = endian != native_endian;
 
+        comptime assert(!@typeInfo(OSABI).@"enum".is_exhaustive);
         const os_abi: OSABI = @enumFromInt(hdr32.e_ident[EI_OSABI]);
 
         // The meaning of this value depends on `os_abi` so just make it available as `u8`.
         const abi_version = hdr32.e_ident[EI_ABIVERSION];
 
-        const @"type" = if (need_bswap) blk: {
+        const @"type": ET = if (need_bswap) blk: {
+            // Converting integers to exhaustive enums using `@enumFromInt` could cause a panic.
+            comptime assert(@typeInfo(ET).@"enum".is_exhaustive);
             const value = @intFromEnum(hdr32.e_type);
-            break :blk @as(ET, @enumFromInt(@byteSwap(value)));
+            break :blk std.meta.intToEnum(ET, @byteSwap(value)) catch |err| switch (err) {
```

Comment:
Now that the code uses [`std.meta.intToEnum`](https://github.com/ziglang/zig/blob/d4c85079c5796805e6f44cadaee618468267db63/lib/std/meta.zig#L885), it should correctly handle values invalid for the destination `enum`, regardless of exhaustiveness.
Is the `assert(is_exhaustive)` still relevant then?

Is it meant to convey that for non-exhaustive `ET`, `std.meta.intToEnum` would be unnecessary?
In that case the code could just cover both cases:
```zig
const value = @byteSwap(@intFromEnum(hdr32.e_type));
if (!exhaustive) break @enumFromInt(...);
break intToEnum(..) catch ...
```

but since the `std.meta.intToEnum` path is valid either way, the early break would only be a manual optimization(?).